src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/cft/query/CftQueryServiceTest.java

```diff
@@ -0,0 +1,1431 @@
+        );
+    }
+
+    private static Stream<TaskQueryScenario> grantTypeSpecificScenarioHappyPath() {
+        SearchTaskRequest searchTaskRequest = new SearchTaskRequest(List.of(
+            new SearchParameter(JURISDICTION, SearchOperator.IN, asList("IA")),
+            new SearchParameter(CASE_ID, SearchOperator.IN, asList(
+                "1623278362431003"
+            ))
+        ));
+
+        final TaskQueryScenario publicClassification = TaskQueryScenario.builder()
+            .scenarioName("specific_grant_type_with_classification_as_public")
+            .firstResults(0)
+            .maxResults(10)
+            .roleAssignments(roleAssignmentsWithGrantTypeSpecific(Classification.PUBLIC))
+            .searchTaskRequest(searchTaskRequest)
+            .expectedSize(1)
+            .expectedTaskDetails(Lists.newArrayList(
+                "8d6cc5cf-c973-11eb-bdba-0242ac111003", "1623278362431003"
+                )
+            ).build();
+
+        final TaskQueryScenario privateClassification = TaskQueryScenario.builder()
+            .scenarioName("specific_grant_type_with_classification_as_private")
+            .firstResults(0)
+            .maxResults(10)
+            .roleAssignments(roleAssignmentsWithGrantTypeSpecific(Classification.PRIVATE))
+            .searchTaskRequest(searchTaskRequest)
+            .expectedSize(1)
+            .expectedTaskDetails(Lists.newArrayList(
+                "8d6cc5cf-c973-11eb-bdba-0242ac111003", "1623278362431003"
+                )
+            ).build();
+
+        final TaskQueryScenario restrictedClassification = TaskQueryScenario.builder()
+            .scenarioName("specific_grant_type_with_classification_as_restricted")
+            .firstResults(0)
+            .maxResults(10)
+            .roleAssignments(roleAssignmentsWithGrantTypeSpecific(Classification.RESTRICTED))
+            .searchTaskRequest(searchTaskRequest)
+            .expectedSize(1)
+            .expectedTaskDetails(Lists.newArrayList(
+                "8d6cc5cf-c973-11eb-bdba-0242ac111003", "1623278362431003"
+                )
+            ).build();
+
+        return Stream.of(
+            publicClassification,
+            privateClassification,
+            restrictedClassification
+        );
+    }
+
+    private static Stream<TaskQueryScenario> grantTypeBasicSpecificErrorScenario() {
+
+        List<RoleAssignment> roleAssignments = new ArrayList<>();
+        final Map<String, String> tcAttributes = Map.of(
+            RoleAttributeDefinition.CASE_TYPE.value(), "Asylum",
+            RoleAttributeDefinition.JURISDICTION.value(), "IA",
+            RoleAttributeDefinition.CASE_ID.value(), "1623278362431003"
+        );
+        RoleAssignment roleAssignment = RoleAssignment.builder().roleName("tribunal-caseworker")
+            .classification(Classification.RESTRICTED)
+            .beginTime(LocalDateTime.now().minusYears(1))
+            .authorisations(List.of("DIVORCE", "PROBATE"))
```

Comment:
do we also need too capture scenario when authorisation is null or empty?

Reply:
Its already covered in HappyPath